orphanvms/_ec2

```diff
@@ -77,6 +78,33 @@ for instance_index in $(seq 1 $(jq -e 'length'<<<"$simple_inst_list")); do
     if inuseby_tag=$(get_tag_value "in-use-by" "$instance"); then
         dbg "Found instance '$name' tagged in-use-by=$inuseby_tag."
         line+=" tagged in-use-by=$inuseby_tag"
+    else  # NOT a persistent=true or a likely get_ci_vm instance
+        # Around Jun/Jul '23 an annoyingly steady stream of EC2 orphans were
+        # reported to Cirrus-support.  They've taken actions to resolve,
+        # but the faulure-modes are many and complex.  Since most of the EC2
+        # instances are rather expensive to keep needlessly running, and manual
+        # cleanup is annoying, try to terminate them automatically.
+        dbg("Attempting to terminate instance $instance (Name=$name)")
+
+        # Operation runs asynchronously, no error reported for already terminated instance.
+        # Any stdout/stderr here would make the eventual e-mail unreadable.
+        if ! termout=$(aws ec2 terminate-instances --no-paginate --output json --instance-ids "$instance" 2>&1)
+        then
+            # Catch rare TOCTOU race, instance was running, terminated, and pruned while looping.
+            # (terminated instances stick around for a while until purged automatically)
+            if [[ "$termout" =~ InvalidInstanceID ]]; then
+                line+=" auto-termination: instance vanished"
```

Comment:
I think a separator would be really nice here, to avoid a run-on:
```diff
- ... running X days auto-termination ETC ...
+ ... running X days; auto-termination ETC ...
```
...but it's probably not worth a re-push.

Reply:
Oh okay, thanks.  If I need to re-push I'll add that.